ibm/service/cos/resource_ibm_cos_replication_configuration.go

```diff
@@ -0,0 +1,438 @@
+package cos
+
+import (
+	"fmt"
+	"strings"
+	"time"
+
+	bxsession "github.com/IBM-Cloud/bluemix-go/session"
+	"github.com/IBM-Cloud/terraform-provider-ibm/ibm/conns"
+	"github.com/IBM-Cloud/terraform-provider-ibm/ibm/flex"
+	"github.com/IBM-Cloud/terraform-provider-ibm/ibm/validate"
+	"github.com/IBM/ibm-cos-sdk-go/aws"
+	"github.com/IBM/ibm-cos-sdk-go/aws/credentials/ibmiam"
+	token "github.com/IBM/ibm-cos-sdk-go/aws/credentials/ibmiam/token"
+	"github.com/IBM/ibm-cos-sdk-go/aws/session"
+	"github.com/IBM/ibm-cos-sdk-go/service/s3"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+
+	validation "github.com/hashicorp/terraform-plugin-sdk/v2/helper/validation"
+)
+
+func ResourceIBMCOSBucketReplicationConfiguration() *schema.Resource {
+	return &schema.Resource{
+		Create:   resourceBucketReplicationConfigurationCreate,
+		Read:     resourceIBMCOSBucketReplicationConfigurationRead,
+		Update:   resourceIBMCOSBucketReplicationConfigurationUpdate,
+		Delete:   resourceIBMCOSBucketReplicationConfigurationDelete,
+		Importer: &schema.ResourceImporter{},
+
+		Timeouts: &schema.ResourceTimeout{
+			Create: schema.DefaultTimeout(60 * time.Minute),
+			Update: schema.DefaultTimeout(20 * time.Minute),
+			Delete: schema.DefaultTimeout(10 * time.Minute),
+		},
+		Schema: map[string]*schema.Schema{
+			"bucket_crn": {
+				Type:        schema.TypeString,
+				Required:    true,
+				ForceNew:    true,
+				Description: "COS bucket CRN",
+			},
+			"bucket_location": {
+				Type:        schema.TypeString,
+				Required:    true,
+				ForceNew:    true,
+				Description: "COS bucket location",
+			},
+			"endpoint_type": {
+				Type:         schema.TypeString,
+				Optional:     true,
+				ValidateFunc: validate.ValidateAllowedStringValues([]string{"public", "private", "direct"}),
+				Description:  "COS endpoint type: public, private, direct",
+				Default:      "public",
+			},
+			"replication_rule": {
+				Type:        schema.TypeList,
+				Required:    true,
+				MaxItems:    1,
+				Description: "Replicate objects between buckets, replicate across source and destination. A container for replication rules can add up to 1,000 rules. The maximum size of a replication configuration is 2 MB.",
+				Elem: &schema.Resource{
```

Comment:
If multiple rules can be added then you need to support updating adding and removing a rule

Reply:
multiple rules are added